src/quick_find.rs

```diff
@@ -7,7 +7,7 @@ use core::panic;
 /// symmetric and transitive.
 /// It's not directly usable when relationships have direction (e.g. if A points to B, B doesn't necessarily points to
 /// A).
-trait UnionFind {
+pub trait UnionFind {
```

Comment:
It would be better to move this trait into a dedicated file, since it is shared by multiple `struct`.